### ISSUE TITLE
could lock be enhanced by StampedLock ?

### ISSUE THREAD

**Issue by bwzhang2011** [jankotek/mapdb#474]
@jankotek , from mapdb, there's a lot of usecases for ReentrantReadWriteLock for write lock and read lock.
as JDK8 or jsr166e brought in the StampedLock  which could help improve for lock speed for single thread or multiple threads use cases. and could such be brought in to support that, maybe mapdb could add more branch to support higher JDK for performance.

**Comment by jankotek** [comment]
I run benchmarks and `StampedLock` was much slower compared to `ReentrantReadWriteLock`. There are cases where upgrade from read lock to write lock would be useful.

Also MapDB depends on JDK6 (android), I have no plans yet to bump minimal version required yet. I seen some project on Google Code. 

For single threaded use one can disable locks completely, it is already in 2.0 branch. 

Not going to implement `StampedLock` in near future.

**Comment by bwzhang2011** [comment]
@jankotek, thanks for reply. It was first time told that StampedLock  much slower than ReentrantReadWriteLock. would you mind leaving me more benchmark test example to testify that ?
I really agree with that the mapdb should keep in mind with android support, but as you know mapdb could catch up with jdk8 in the future plan (just some suggestion). you mentioned that we could disable the lock completely, how could I enable that through the configuration ?

**Comment by bwzhang2011** [comment]
@jankotek, thanks for further investigation. such similar project is also forked in github. please pay a visit to 
https://github.com/manuelroth/concurrent-locks.

**Comment by bwzhang2011** [comment]
@jankotek, any update with the issue ? any idea with the concurrent-lock brought in for the original lock implementation replacement ?

**Comment by jankotek** [comment]
Hi, it is low priority new feature, I will investigate in a few months.

**Comment by bwzhang2011** [comment]
@jankotek, I will close it as you promise to improve the locking performance. for my test is focus on the integration with hazelcast and mapdb for hz3.4+, I noticed that it degrade most when large data sets brought in,
one is for the direct speed between heap and off-heap another is for lock processing, so looking forward to your improvement for mapdb2.0 stable and also the mapdb-hz together is much more appreciated of.

**Comment by jankotek** [comment]
Leave it open, it is good reminder

**Comment by jankotek** [comment]
Hi, MapDB 2.1 will depend on Java 8. Java 6 version will be provided with code preprocessors. 

Closing this issue as StampedLock are on my todo list for 2.1

**Comment by dmk23** [comment]
A slightly related question- is MapDB 2.0 fully tested / supported on JDK 8?

**Comment by jankotek** [comment]
I tested on OpenJDK7. I will run Java8 tests in near future.

Jan

On Friday, July 10, 2015 21:57:43 dmk23 wrote:


A slightly related question- is MapDB 2.0 fully tested / supported on JDK 8? 
—Reply to this email directly or view it on GitHub[1].[2]




--------
[1] https://github.com/jankotek/mapdb/issues/474#issuecomment-120572879
[2] 
https://github.com/notifications/beacon/AAV6bbiYTwHWnhl8pZgO3Bsrh7jEdzhtks5
ocJnHgaJpZM4D4Re7.gif

**Comment by dmk23** [comment]
Is this something that could be prioritized? Seems to me if 2.1 is intended to *require* Java 8, it should be natural for the immediate prior release to be fully supported on JDK 8...

**Comment by jankotek** [comment]
No, 2.1 will have some support for Java6 in different package (mapdb-jdk6). Code 
which will depend on Java8 has not been written yet, so there is no reason to test 
it.  

On Saturday, July 11, 2015 00:25:05 dmk23 wrote:


Is this something that could be prioritized? Seems to me if 2.1 is intended to 
/require/ Java 8, it should be natural for the immediate prior release to be fully 
supported on JDK 8... 
—Reply to this email directly or view it on GitHub[1].[2]




--------
[1] https://github.com/jankotek/mapdb/issues/474#issuecomment-120589577
[2] https://github.com/notifications/beacon/AAV6bSoVAWdPyq5wkqP_adeL-ORULWQnks5ocLxRgaJpZM4D4Re7.gif

**Comment by dmk23** [comment]
This reason would be because many people are moving everything to JDK8 because of requirements from other software they are using... So I'd say any software project intended for wide commercial adoption these days should be making JDK8 support a priority...
